lucille-plugins/lucille-jlama/src/main/java/com/kmwllc/lucille/jlama/stage/JlamaEmbed.java

```diff
@@ -0,0 +1,131 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import org.apache.zookeeper.common.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This stage uses jlama as an inference machine to download and use the embedding model locally, where it would retrieve its input
+ * from a source field, and set the outputs to a destination field.
+ *
+ * Config Parameters:
+ * - source (String) : field of which the embedding Stage will retrieve content from
+ * - dest (String, Optional) : name of the field that will hold the embeddings, defaults to "embeddings"
+ * - localModelFolderPath (String) : Path where the model will be downloaded to and used
+ * - model (String, Optional) : the name of the embedding model to use in the format of owner/name
+ * - workingMemoryType (String, Optional) : the data type used during runtime for model, impacts accuracy and resource usage.
+ * - workingQuantizationType (String, Optional): the data type used during quantization for model, influences the compression and
+ *   speed of the model, trading off accuracy for performance.
+ *
+ *   DType Options:
+ *    "BOOL" (1 byte)
+ *    "U8" (Unsigned 8-bit integer)
+ *    "I8" (Signed 8-bit integer)
+ *    "I16" (Signed 16-bit integer)
+ *    "U16" (Unsigned 16-bit integer)
+ *    "F16" (16-bit floating point)
+ *    "BF16" (BFloat16, 16-bit floating point)
+ *    "I32" (Signed 32-bit integer)
+ *    "U32" (Unsigned 32-bit integer)
+ *    "F32" (32-bit floating point)
+ *    "F64" (64-bit floating point)
+ *    "I64" (Signed 64-bit integer)
+ *    "U64" (Unsigned 64-bit integer)
+ *    "Q4" (Quantized 4-bit integer)
+ *    "Q5" (Quantized 5-bit integer)
+ */
+
+public class JlamaEmbed extends Stage {
+  public final String embeddingModel;
+  public final String source;
+  public final String pathToStoreModel;
+  public final String workingMemoryType;
+  public final String workingQuantizationType;
+  public final String dest;
+  public AbstractModel model;
+
+  private static final Logger log = LoggerFactory.getLogger(OpenAIEmbed.class);
+
+  public JlamaEmbed(Config config) {
+    super(config, new StageSpec()
+        .withRequiredProperties("model", "source", "pathToStoreModel")
+        .withOptionalProperties("workingMemoryType", "workingQuantizationType"));
+    pathToStoreModel = config.getString("pathToStoreModel");
+    this.embeddingModel = config.getString("model");
+    this.source = config.getString("source");
+    this.dest = config.hasPath("dest") ? config.getString("dest") : "embeddings";
+    this.workingMemoryType = config.hasPath("workingMemoryType") ? config.getString("workingMemoryType") : "";
+    this.workingQuantizationType = config.hasPath("workingQuantizationType") ? config.getString("workingQuantizationType") : "";
+  }
+
+  @Override
+  public void start() throws StageException {
+    try {
+      File localModelPath = SafeTensorSupport.maybeDownloadModel(pathToStoreModel, embeddingModel);
```

Comment:
What do folks think about a "removeModelAfter" boolean stage property that removes the model folder using the path that it was stored at in the stop() method of the stage?